spec/models/email_settings_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'spec_helper'
+
+describe EmailSettings do
+  let(:conference) { create(:conference, short_title: 'goto', start_date: Date.new(2014, 05, 01), end_date: Date.new(2014, 05, 06)) }
+  let(:user) { create(:user, email: 'john@doe.com', name: 'John Doe') }
+  let(:event_user) { create(:submitter, user: user) }
+  let(:event) { create(:event, program: conference.program, title: 'Talk about talks', event_users: [event_user]) }
+  let(:expected_hash) do
+    {
+      'email' => 'john@doe.com',
+      'name' => 'John Doe',
+      'conference' => 'The dog and pony show',
+      'conference_start_date' => Date.new(2014, 05, 01),
+      'conference_end_date' => Date.new(2014, 05, 06),
+      'registrationlink' => 'http://localhost:3000/conference/goto/register',
+      'conference_splash_link' => 'http://localhost:3000/conference/goto',
+      'schedule_link' => 'http://localhost:3000/conference/goto/schedule',
+      'cfp_end_date' => 'Unknown',
+      'cfp_start_date' => 'Unknown',
+      'venue' => 'Unknown',
+      'venue_address' => 'Unknown'
+    }
+  end
+
+  it 'has a valid factory' do
+    expect(build(:email_settings)).to be_valid
+  end
+
+  describe '#get_values' do
+    it 'returns correct key-value pairs' do
+      expect(conference.email_settings.get_values(conference, user)).to eq expected_hash
+    end
+
+    context 'conference has cfp' do
+      before do
+        conference.program.update_attributes(cfp: create(:cfp,
+                                                         start_date: Date.new(2014, 04, 29),
+                                                         end_date: Date.new(2014, 05, 06)))
+      end
+
+      it 'returns hash with cfp start and end date' do
+        cfp_dates_hash = { 'cfp_start_date' => Date.new(2014, 04, 29), 'cfp_end_date' => Date.new(2014, 05, 06) }
```

Comment:
You can move this also to the before block 
``
cfp_dates_hash = { 'cfp_start_date' => Date.new(2014, 04, 29), 'cfp_end_date' => Date.new(2014, 05, 06) }
expected_hash.merge!(cfp_dates_hash)
``